docs/design/dedicated-volume.md

```diff
@@ -0,0 +1,115 @@
+---
+title: Dedicated Volumes
+authors:
+  - "@hzxuzhonghu"
+reviewers:
+  - "@k82cn"
+  - "@zrss"
+approvers:
+  - "@k82cn"
+creation-date: 2019-11-21
+last-updated: 2019-11-21
+status: review
+---
+
+# Dedicated Volumes
+
+## Summary
+
+The purpose of this is to allow mount dedicated volumes per pod of a Volcano Job.
+
+## Motivation
+
+Volume mount was supported from begin. But there are few limitations:
+
+1. We can specify volumes of Job scope, all the volumes are shared by all pods within a Job.
+
+2. We specify volumes by setting `TaskSpec.PodTemplateSpec.Volumes`, but similarly they are shared by pods within a task.
+
+But in real world, scenarios like DL/BigData, etc requires high performance. Shared storage has some performance issue, 
+like io limit, read/write conflicts. 
+
+Also some cloud vendors do not support volumes mounting to multiple nodes, it is to prevent data inconsistent.
+
+Given the above issues, we should support pods within volcano mounting volumes exclusively.
+
+## Goals
+
+Job TaskSpec will contain an additional structure called `VolumeClaimTemplates` to control exclusive volumes.
+
+## Proposal
+
+### Implementation Details
+
+#### API Changes
+
+Following changes will be made to the TaskSpec for volcano Job.
+
+```go
+// TaskSpec specifies the task specification of Job
+type TaskSpec struct {
+    ...
+
+    // volumeClaimTemplates is a list of claims that pods are allowed to reference.
+    // The Job controller is responsible for mapping network identities to
+    // claims in a way that maintains the identity of a pod. Every claim in
+    // this list must have at least one matching (by name) volumeMount in one
+    // container in the template. A claim in this list takes precedence over
+    // any volumes in the template, with the same name.
+    // +optional
+    VolumeClaimTemplates []v1.PersistentVolumeClaim `json:"volumeClaimTemplates,omitempty" protobuf:"bytes,5,rep,name=volumeClaimTemplates"`
+
+    ...
+}
+```
+
+- By default, this is empty. Will use that defined in `JobSpec` and `TaskSpec.Template`.
+
+- If `VolumeClaimTemplates` specified, these pvcs are referenced by all the pods of the task. But the persistent volume claim name is rebuilt by job controller. In a word, pods will not share any of the persistent volume claims. 
+
+- If the rebuilt persistent does not exist, job controller will create it.
+
+
+#### Implementation
+
+Consider the following demo job:
+
+```yaml
+apiVersion: batch.volcano.sh/v1alpha1
+kind: Job
+metadata:
+  name: demo
+spec:
+  tasks:
+  - name: task
+    replicas: 2
+    template:
+      spec:
+        containers:
+        - name: ps
+          image: ps-img
+          volumeMounts:
+          - mountPath: /data
+            name: train-vol
+    volumeClaimTemplates:
+    - metadata:
+        name: train-vol
```

Comment:
if using `GenerateName`, we'll generate pvc for each pods. if using `name`, only one pvc for all pods.

Reply:
@hzxuzhonghu , is that ok for you?

Reply:
Agree, i will update to explicitly indicate it.